packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -108,32 +106,77 @@ export type SecurityProviderRequest = (
   messageType: string,
 ) => Promise<Json>;
 
-// TODO: Either fix this lint violation or explain why it's necessary to ignore.
-// eslint-disable-next-line @typescript-eslint/naming-convention
-type getCurrentChainId = () => Hex;
+type AbstractMessageManagerOptions<
+  Message extends AbstractMessage,
+  Action extends ActionConstraint,
+  Event extends EventConstraint,
+> = {
+  additionalFinishStatuses?: string[];
+  messenger: RestrictedControllerMessenger<
+    string,
+    Action,
+    Event,
+    string,
+    string
+  >;
+  name: string;
+  securityProviderRequest?: SecurityProviderRequest;
+  state?: MessageManagerState<Message>;
+};
 
 /**
  * Controller in charge of managing - storing, adding, removing, updating - Messages.
  */
 export abstract class AbstractMessageManager<
-  // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  M extends AbstractMessage,
-  // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  P extends AbstractMessageParams,
-  // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  PM extends AbstractMessageParamsMetamask,
-> extends BaseControllerV1<BaseConfig, MessageManagerState<M>> {
-  protected messages: M[];
-
-  protected getCurrentChainId: getCurrentChainId | undefined;
+  Message extends AbstractMessage,
+  Params extends AbstractMessageParams,
+  ParamsMetamask extends AbstractMessageParamsMetamask,
+  Action extends ActionConstraint,
+  Event extends EventConstraint,
+> extends BaseController<
+  string,
+  MessageManagerState<Message>,
+  RestrictedControllerMessenger<string, Action, Event, string, string>
+> {
+  protected messages: Message[];
 
   private readonly securityProviderRequest: SecurityProviderRequest | undefined;
 
   private readonly additionalFinishStatuses: string[];
 
+  hub: EventEmitter = new EventEmitter();
+
+  /**
+   * Creates an AbstractMessageManager instance.
+   *
+   * @param options - Options for the AbstractMessageManager.
```

Comment:
Rather than defining this JSDoc on the constructor, would it be easier to define it on the options type and also export it?